dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/Services/UserImport/UserImportProcessorTests.cs

```diff
@@ -0,0 +1,1057 @@
+        });
+    }
+
+    [Fact]
+    public async Task Process_WithCsvRowWithMoreFieldsThanExpected_InsertsUserImportJobRowWithErrorIntoDatabase()
+    {
+        // Arrange
+        var dbContext = _dbFixture.GetDbContext();
+        var userImportStorageService = new Mock<IUserImportStorageService>();
+        var clock = new TestClock();
+        var logger = new Mock<ILogger<UserImportProcessor>>();
+        var userImportJobId = Guid.NewGuid();
+        var storedFilename = $"{userImportJobId}.csv";
+        var csvContent = new StringBuilder();
+        csvContent.AppendLine("ID,EMAIL_ADDRESS,FIRST_NAME,LAST_NAME,DATE_OF_BIRTH");
+        csvContent.AppendLine("UserImportProcessorTests3,UserImportProcessorTests3@email.com,joe,bloggs,05021970,extra,data");
+        using var csvStream = new MemoryStream(Encoding.UTF8.GetBytes(csvContent.ToString()));
+
+        await _dbFixture.TestData.WithDbContext(async dbContext =>
+        {
+            dbContext.UserImportJobs.Add(new UserImportJob()
+            {
+                UserImportJobId = userImportJobId,
+                StoredFilename = storedFilename,
+                OriginalFilename = "my-user-import.csv",
+                UserImportJobStatus = UserImportJobStatus.New,
+                Uploaded = clock.UtcNow
+            });
+
+            await dbContext.SaveChangesAsync();
+        });
+
+        userImportStorageService.Setup(s => s.OpenReadStream(storedFilename))
+            .ReturnsAsync(csvStream)
+            .Verifiable();
+
+        // Act
+        var userImportProcessor = new UserImportProcessor(
+            dbContext,
+            userImportStorageService.Object,
+            clock,
+            logger.Object);
+        await userImportProcessor.Process(userImportJobId);
+
+        // Assert
+        userImportStorageService
+            .Verify();
+
+        await _dbFixture.TestData.WithDbContext(async dbContext =>
+        {
+            var user = await dbContext.Users.SingleOrDefaultAsync(r => r.EmailAddress == "UserImportProcessorTests3@email.com");
+            Assert.Null(user);
+
+            var events = await dbContext.Events.Where(e => e.EventName == "UserImportedEvent").ToListAsync();
+            var userImportedEventCount = events
+                .Select(e => JsonSerializer.Deserialize<UserImportedEvent>(e.Payload))
+                .Count(i => i!.UserImportJobId == userImportJobId);
+            Assert.Equal(0, userImportedEventCount);
+
+            var userImportJobRow = await dbContext.UserImportJobRows.SingleOrDefaultAsync(r => r.UserImportJobId == userImportJobId);
+            Assert.NotNull(userImportJobRow);
+            Assert.NotNull(userImportJobRow.Errors);
+            Assert.Single(userImportJobRow.Errors);
+            Assert.Equal($"Exactly {UserImportRow.ColumnCount} fields expected (this row has more)", userImportJobRow.Errors[0]);
+        });
+    }
+
+    [Fact]
```

Comment:
The tests for validation errors look very similar; perhaps a single `[Theory]` with a case for each error?